java-manta-client/src/main/java/com/joyent/manta/util/ConcurrentWeakIdentityHashMap.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
@cburroughs Can you provide some guidance on this? I'm a bit new to working with the MPLv2.